pyproject.toml

```diff
@@ -1,3 +1,114 @@
+[project]
+name = "ggshield"
+dynamic = ["version"]
+description = "Detect secrets from all sources using GitGuardian's brains"
+keywords = [
+    "cli",
+    "devsecops",
+    "secrets-detection",
+    "security-tools",
+    "gitguardian",
+]
+readme = "README.md"
+authors = [
+    {name = "GitGuardian", email = "support@gitguardian.com"},
+]
+maintainers = [
+    {name = "GitGuardian", email = "support@gitguardian.com"},
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: End Users/Desktop",
+    "Natural Language :: English",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: Console",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Operating System :: OS Independent",
+    "Topic :: Security",
+]
+license = {text = "MIT"}
+requires-python = ">=3.8.1"
+dependencies = [
+    "platformdirs>=3.0.0,<3.1.0",
+    "charset-normalizer>=3.1.0,<3.2.0",
+    "click>=8.1,<8.2",
+    "cryptography>=43.0.1,<44.0.0",
+    "marshmallow>=3.18.0,<3.19.0",
+    "marshmallow-dataclass>=8.5.8,<8.6.0",
+    "oauthlib>=3.2.1,<3.3.0",
+    "pygitguardian>=1.17.0,<1.18.0",
+    "pyjwt>=2.6.0,<2.7.0",
+    "python-dotenv>=0.21.0,<0.22.0",
+    "pyyaml>=6.0.1,<6.1",
+    "requests>=2.32.0,<2.33.0",
+    "rich>=12.5.1,<12.6.0",
+    "typing-extensions>=4.12.2,<5.0.0",
+    "urllib3>=2.2.2,<2.3.0",
+]
+
+[project.urls]
+Homepage = "https://github.com/GitGuardian/ggshield"
+
+[project.scripts]
+ggshield = "ggshield.__main__:main"
+
+[build-system]
+requires = ["pdm-backend"]
+build-backend = "pdm.backend"
+
+[tool.pdm.dev-dependencies]
+dev = [
+    "scriv[toml]",
+    "black==24.3.0",
+    "coverage",
+    "flake8",
+    "flake8-isort",
+    "flake8-quotes",
+    "pre-commit",
+    "pyright==1.1.367",
+    "build>=1.2.2.post1",
+]
+
+tests = [
+    # pinning because of conflicting dependencies with exceptiongroup",
+    "pytest==7.2.1",
+    "pytest-mock",
+    "pytest-socket",
+    "pytest-voluptuous",
+    "seed-isort-config",
+    "snapshottest",
+    "import-linter",
+    "jsonschema",
+    "vcrpy>=5.1.0",
+    # Dependency of pytest-voluptuous. Pin it because 0.15.0 does not",
+    # support Python 3.8 anymore.",
+    "voluptuous<0.15.0",
+    # Pin because 5.6.0 currently (2024-07-15) does not work on Windows",
+    "pyfakefs>=5.2.0,<5.6.0",
```

Comment:
this "currently" is 5 month old and many several versions have been released meanwhile, and even so, the [5.2.2](https://pypi.org/project/pyfakefs/5.2.3/) claims to work on Windows, so I don't get it

Reply:
This comment does not say 5.2.2 does not work on Windows. It says 5.6 does not.